spec/data-model.md

```diff
@@ -145,17 +145,22 @@ interface Option {
 }
 ```
 
-A `Reserved` represents an _expression_ with a _reserved_ _annotation_.
-The `sigil` corresponds to the starting sigil of the _reserved_.
+A `Reserved` represents an _expression_ with a _reserved_ or _private-use_ _annotation_.
+The `sigil` corresponds to the starting sigil of the _annotation_.
 The `source` is the "raw" value (i.e. escape sequences are not processed)
-and includes the starting `sigil`.
+and does not include the starting `sigil`.
 
 Implementations MUST NOT rely on the set of `sigil` values remaining constant,
 as future versions of this specification MAY assign other meanings to such sigils.
 
 If the _expression_ includes a _literal_ or _variable_ before the _annotation_,
 it is included as the `operand`.
 
+When parsing the syntax of a _message_ that includes a _private-use_ _annotation_
+supported by the implementation,
+the implemenation MAY represent it in the data model using a different interface
```

Comment:
I think we might need some stronger guidance here. I agree with the "MAY" you have, but would suggest something along the lines of:

> _Private-use_ _annotations_ are specific to a particular "private agreement"
> between the various implementations that support a given form of _private-use_.
>
> When parsing the syntax of a _message_ that includes a _private-use_ _annotation_
> unrecognized by the implementation, the _annotation_ MUST be processed
> identically to a _reserved_ _annotation_.
>
> An implementation that supports a given _private-use_ _annotation_ MUST
> define the specific interface to support the semantics, structure, and meaning
> that it provides. Use of existing data model interfaces is RECOMMENDED, although
> an implementation MAY use any interface appropriate for its needs.

Reply:
I don't think the "parsing the syntax" sentence belongs in the data model spec, but in the syntax. It's defining a MUST about what the syntax means, and if included here that would get watered down by the qualifier at the top of this doc:
> Implementations are not required to use this data model for their internal representation of messages.

On review, I now also note that the Extensions section will need to be updated to account for private-use. Something like your third paragraph is probably needed there.